Sources/SpeziOnboarding/OnboardingFlow/OnboardingViewController.swift

```diff
@@ -0,0 +1,40 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import SwiftUI
+
+public class OnboardingViewController: ObservableObject {
```

Comment:
I would suggest that the state management object works using a `NavigationPath` that is part of the SwiftUI API surface: https://developer.apple.com/documentation/swiftui/navigationpath

The following example provides a good overview how Apple models a `ObservableObject` around this: https://developer.apple.com/documentation/swiftui/navigationpath#Serialize-the-path ... the serialization might be a good feature as well.